src/runtime/BackingStore.cpp

```diff
@@ -50,21 +64,40 @@ BackingStore::BackingStore(void* data, size_t byteLength, BackingStoreDeleterCal
                                    nullptr, nullptr, nullptr);
 }
 
-BackingStore::BackingStore(size_t byteLength)
-    : BackingStore(calloc(byteLength, 1), byteLength,
-                   [](void* data, size_t length, void* deleterData) {
-                       free(data);
-                   },
-                   nullptr)
+BackingStore::BackingStore(VMInstance* instance, size_t byteLength)
+    : BackingStore(instance->platform()->onMallocArrayBufferObjectDataBuffer(byteLength), byteLength,
+                   defaultPlatformBackingStoreDeleter, instance, false, true)
 {
 }
 
+void BackingStore::reallocate(VMInstance* instance, size_t newByteLength)
+{
+    // Shared Data Block should not be reallocated
+    ASSERT(!m_isShared);
+    if (m_isAllocatedByPlatformAllocator) {
+        m_data = instance->platform()->onReallocArrayBufferObjectDataBuffer(m_data, m_byteLength, newByteLength);
+        m_byteLength = newByteLength;
+    } else {
+        m_deleter(m_data, m_byteLength, m_deleterData);
+        m_data = instance->platform()->onMallocArrayBufferObjectDataBuffer(newByteLength);
+        m_deleter = defaultPlatformBackingStoreDeleter;
+        m_deleterData = instance;
+        m_byteLength = newByteLength;
+        m_isAllocatedByPlatformAllocator = true;
```

Comment:
After calling this, why should it always change to the m_isAllocatedByPlatformAllocator true state? even if ```newByteLength``` is equal to ```m_byteLength```?

Reply:
We should change m_isAllocatedByPlatformAllocator to true.
because we can not know how user allocated memory if m_isAllocatedByPlatformAllocator==false
BTW, do not any operation if m_byteLength==newByteLength is good idea

Reply:
For the case that BackingStore is allocated previously by user-defined manner, new memory is allocated through PlatformAllocator(`Platform::onMallocArrayBufferObjectDataBuffer`) for reallocate operation. Is it desired? I mean, we don't need to consider the user-defined reallocate operation here?

Reply:
We don't store allocation callback in BackingStore.
IMO store the callback is don't needed for BackingStore.
reallocation is not famous API and If user want to his allocator, user can make use BackingStore.
and I don't want to allow failure when user reallocate BackingStore